python/python/lance/dataset.py

```diff
@@ -894,9 +899,15 @@ def create_index(
             Replace the existing index if it exists.
         num_partitions : int, optional
             The number of partitions of IVF (Inverted File Index).
-        ivf_centroids : ``np.ndarray`` or ``pyarrow.FixedSizeListArray``. Optional.
+        ivf_centroids : ``np.ndarray``, ``pyarrow.FixedSizeListArray``
+        or ``pyarrow.FixedShapeTensorArray``. Optional.
             A ``num_partitions x dimension`` array of K-mean centroids for IVF
             clustering. If not provided, a new Kmean model will be trained.
+        pq_codebook : ``np.ndarray``, ``pyarrow.FixedSizeListArray``
+        or ``pyarrow.FixedShapeTensorArray``. Optional.
+            A ``num_sub_vectors x (256 * dimensions // num_sub_vectors)``
+            array of K-mean centroids for PQ codebook.
```

Comment:
Wanna to add a comment that this is residual PQ? Now i remember that one reason that we did not allow user to pass PQ is that this might be confusing for users to know what is residual pq or not.

Also, lets make `256` to presented as `2 ^ nbits`, we can leave room for other `nbits`